src/compiler/types.ts

```diff
@@ -831,6 +831,10 @@ export const enum NodeFlags {
     // never cleared on SourceFiles which get re-used in between incremental parses.
     // See the comment above on `PossiblyContainsDynamicImport` and `PossiblyContainsImportMeta`.
     /** @internal */ PermanentlySetIncrementalFlags = PossiblyContainsDynamicImport | PossiblyContainsImportMeta,
+
+    // The following flags repurpose other NodeFlags as different meanings for Identifier nodes
+    /** @internal */ IdentifierHasExtendedUnicodeEscape = ContainsThis, // Indicates whether the identifier contains an extended unicode escape sequence
+    IdentifierIsInJSDocNamespace = HasAsyncFunctions, // Indicates whether the identifier is part of a JSDoc namespace

```

Comment:
I would bet you could make this `/** @internal */` without anyone being mad. Looks like practically speaking, nobody uses this except clones of us and codebases adapting our tree shapes.

https://github.com/search?q=isInJSDocNamespace+NOT+path%3A%2F%5Esrc%5C%2Fcompiler%5C%2F%2F&type=code